.github/workflows/test-build.yml

```diff
@@ -0,0 +1,70 @@
+name: Build tests
+on: [pull_request]
+jobs:
+
+  kirkstone-repo:
+    runs-on: self-hosted
+    if: "!contains(github.event.head_commit.message, 'ci skip')"
+    steps:
+      - uses: actions/checkout@v3
+      - uses: ./.github/actions/bitbake-repo
+        with:
+          repo_release: 'kirkstone'
+
+  matrix-build-image:
+    strategy:
+      fail-fast: false
+      matrix:
+        machine: ['raspberrypi3-mesa', 'raspberrypi4-64-mesa']
+        wpe_vers: ['2_38', 'nightly']
+        yocto_rel: ['kirkstone']
+    continue-on-error: true
+    runs-on: self-hosted
+    if: "!contains(github.event.head_commit.message, 'ci skip')"
+    needs: kirkstone-repo
+    steps:
+      #- uses: actions/checkout@v3
+      - uses: ./.github/actions/bitbake-build
+        with:
+          bitbake_source: 'raspberrypi3-mesa-wpe-${{ matrix.wpe_vers }} ${{ matrix.machine }} poky-wayland layers.raspberrypi conf.wpe-${{ matrix.wpe_vers }} --update-config'
+          repo_release: ${{ matrix.yocto_rel }}
+
+  # kirkstone-raspberrypi3-mesa-weston-wpe-nightly:
```

Comment:
In principle nope, I started with these and then moved them to be executed via "matrix", and in principle with "fail-fast: false" and "continue-on-error: true" we get the execution for all combinations, but I was thinking about leaving the above as example of things done in a different, more straightforward way

For example, if we want to change invocations or other stuff with WPE 2.40, it might be more easy to keep matrix for the old and the individual jobs for the new, and so we have the example and don't have to dig again in documentation.

But dunno, if you want to keep it clean, we can remove all that.